include/swift/Sema/ConstraintSystem.h

```diff
@@ -5024,6 +5024,24 @@ class OpenUnboundGenericType {
   }
 };
 
+class HandlePlaceholderType {
+  ConstraintSystem &cs;
+  const ConstraintLocatorBuilder &locator;
```

Comment:
I'd suggest we allocate a locator here instead of referencing a builder since we'd have to build a locator per placeholder anyway.

Reply:
Gotcha, will update!